src/util/InstallDriver.ts

```diff
@@ -495,10 +498,48 @@ class InstallDriver {
     this.mApi.dismissNotification(getUnfulfilledNotificationId(collection.id));
     this.mApi.store.dispatch(actions.setModEnabled(profile.id, collection.id, true));
 
-    const required = (collection?.rules ?? [])
+    const required = []
       .filter(rule => rule.type === 'requires');
-    this.mRequiredMods = required
-      .filter(rule => util.findModByRef(rule.reference, mods) === undefined);
+    for (const rule of (collection?.rules ?? [])) {
+      if (rule.type !== 'requires') {
+        continue;
+      }
+
+      const modByRef = util.findModByRef(rule.reference, mods);
+      if (modByRef === undefined) {
+        required.push(rule);
+        continue;
+      }
+
+      if (rule.fileList?.length ?? 0 > 0) {
+        const stagingPath = selectors.installPathForGame(state, profile.gameId);
+        const modPath = path.join(stagingPath, modByRef.installationPath);
+        const presentFiles = await new Promise(async (resolve, reject) => {
+          let entries: IEntry[];
+          await turbowalk(modPath, async input => {
+            entries = [].concat(entries, input);
+          }, {});
+
+          const presentFiles = await Promise.all(entries
+            .filter(iter => !!iter && !iter.isDirectory)
+            .map(async iter => ({
+              path: path.relative(modPath, iter.filePath),
+              md5: await fileMD5Async(iter.filePath),
```

Comment:
This is potentially going to take sooooo long. A user might wait minutes for this to complete, we can't just do this without a progress indicator.